### PR TITLE
RavenDB-15022 documented changes to JSON serialization

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

**Comment by ppekrol** [comment]
Still need to list the differences between the APIs

File: Documentation/5.0/Raven.Documentation.Pages/migration/client-api/conventions.markdown [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Migration: Changes in Conventions
+
+In 5.0 we have introduced an abstract layer over the whole JSON serialization process. This will allow us in the future to introduce more JSON serialization options than currently supported Json.NET one.
+
```
Comment (ayende) [inline Documentation/5.0/Raven.Documentation.Pages/migration/client-api/conventions.markdown:4]: Mention that as part of that, we were able to reduce client side serialization costs by 50%

File: Documentation/5.0/Raven.Documentation.Pages/migration/client-api/conventions.markdown [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+New namespace was introduced, the `Raven.Client.Json.Serialization` containing following classes and interfaces needed for the serialization process:
+
+- `ISerializationConventions` - entry point for the JSON serialization process. Set by default to `JsonNetSerializationConventions`.
+- `IJsonSerializer`, `IJsonWriter`, `IJsonReader` - control the in and out JSON serialization
+- `IBlittableJsonConverter` - allows to convert from entity to blittable and the other way around
+
+{PANEL/}
+
+
+{PANEL:Raven.Client.Json.Serialization.JsonNet namespace}
```
Comment (ayende) [inline Documentation/5.0/Raven.Documentation.Pages/migration/client-api/conventions.markdown:16]: Maybe we should call it NewtonSoft?

File: Documentation/5.0/Raven.Documentation.Pages/migration/client-api/conventions.markdown (hunk 1 above)
Comment (ppekrol) [reply Documentation/5.0/Raven.Documentation.Pages/migration/client-api/conventions.markdown:4]: Done

File: Documentation/5.0/Raven.Documentation.Pages/migration/client-api/conventions.markdown (hunk 2 above)
Comment (ppekrol) [reply Documentation/5.0/Raven.Documentation.Pages/migration/client-api/conventions.markdown]: Half of packages in NuGet have JsonNet, the other half NewtonsoftJson. Went with NewtonsoftJson, this is what MS have.

**Pull request merged**